### DISCUSSION TITLE
Is TouchArea multitouch capable?

### DISCUSSION THREAD

**Discussion by larry104** [slint-ui/slint#2504] [Q&A] [answered]
I am evaluating if I port my application running on a raspberry pi with a touchscreen which consists of a bunch of sliders and keys over to slint. Since I need to be able to move multiple slider at the same time I was wondering if the buildin element TouchArea is multi touch capable? QML has the type MultiPointTouchArea for that purpose which I am currently using.  If not will individual elements fire off touch events even if multiple clicked at the same time?

Any advise?

**Comment by ogoffart** [comment, marked as answer]
right now TouchArea only can do single touch.  Multi touch is supported by the underlying backends do this is just a question of exposing it to the API. 
What exactly are you trying to achieve? Are you trying to recognize a gesture or is it something different.

>  If not will individual elements fire off touch events even if multiple clicked at the same time?

Right now they are not, but this might be a good idea to implement.

**Reply by lambdachad** [reply]
Is this still the case? We would like to have multi touch zoom and panning on mobile phones and tablets, is there a way to do this in Slint currently?

**Reply by ogoffart** [reply]
This is currently not implemented yet. Tracking issue in https://github.com/slint-ui/slint/issues/6704

**Comment by larry104** [comment]
No, I am not trying to recognize gestures, it's just that multiple keys or sliders can be operated at the same time. In QML the only way to do this is by defining a MultiPointTouchArea over the entire screen and when a onPressed/onReleased event occurs iterate over all elements and check if the x/y of the touch point is within the bounds of the element. This works but if it would be implemented in a way that elements can simultaneously fire off events themselves would be way better. QML does not support that as elements only respond to mouse events and since there is only one mouse the is only one event no matter how many points you touch on the screen. Most touch screens support up to 10 touch points. 

Do you think this can be implemented anytime soon since I would really like to switch to slint ;-) 

Do you want me to post this as an issue so it can be added and tracked as improvement?
